feat: add third-party block support

Naccomy opened the pull request:
### Summary

Add methods to create and inspect third party blocks:
- `Biscuit::append_third_party` for creating biscuits with third party blocks
- `Biscuit::external_key` for retrieving external keys from blocks
- Add unit tests covering third party block functionality

divarvel commented:
You’ll need to sign the Eclipse Contributor Agreement: https://github.com/eclipse-biscuit/biscuit-rust/blob/main/CONTRIBUTING.md#eclipse-contributor-agreement

Naccomy commented:
> Neat! A good addition would be to exposing support for requests, since in most cases, we don’t want to transmit the whole token to the third-party.

I’ve updated the code to include `ThirdPartyRequest` and `ThirdPartyBlock`. The `Biscuit::append_third_party` function is now implemented to match the behaviour of the Rust version.

divarvel commented:
I’ll add pyi bindings myself.
